### ISSUE TITLE
When creating credentials inside combustion, jeos-firstboot always called

### ISSUE THREAD

**Issue by aplanas** [openSUSE/jeos-firstboot#128]
When a combustion script creates a systemd-credential, jeos-firstboot is always called even when the file `/sysroot/var/lib/YaST2/reconfig_system` is removed.

I do not understand the root cause, as the combustion script is executed inside the initrd, and the check for the file in the jeos-firstboot.service is happening after initrd.

After changing the combustion script to store the trace logs in /run, I can see that there is always a call to `rm -f /sysroot/var/lib/YaST2/reconfig_system`.

A minimal combustion reproducer is:

```bash
#!/bin/bash
# combustion: network prepare
if [ "${1-}" = "--prepare" ]; then
    exit 0
fi
systemd-machine-id-setup
mount /var
systemd-creds setup
umount /var
exit 0
```

Commenting the `systemd-creds` call will make the jeos service not to run.  The systemd-creds should be creating a 4k random number in var, prefixed with `systemd-id128 machine-id -a d3acecba0dad4cdfb8c9381528936c58`. Not much more, but the part that I a missing somehow makes the [ConditionPathExists=/var/lib/YaST2/reconfig_system](https://github.com/openSUSE/jeos-firstboot/blob/master/files/usr/lib/systemd/system/jeos-firstboot.service#L12) to be accepted.

I write the issue here because a workaround is to move the ConditionPathExist from the service into the code.

**Comment by aplanas** [comment]
Hitting openQA soon

**Comment by mloviska** [comment]
I have removed the `reconfig_system` file just before `umount /var`. Then `jeos-firstboot` is skipped.

**Comment by mloviska** [comment]
> That `umount /var` will cause breakage if this is run with non-transactional systems like Minimal-VM.

Yes, that is true. It only happens in case of tr-up https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/data/microos/butane/script#L118

**Comment by aplanas** [comment]
Should be possible to use `rpm --query --quiet read-only-root-fs` to detect this case, right?

**Comment by Vogtinator** [comment]
> Should be possible to use rpm --query --quiet read-only-root-fs to detect this case, right?

IMO the best possible fix is to address the root cause, which is that it unmounted a previously mounted /var:

```
umount_var=0
if ! mountpoint -q /var; then
    mount /var
    umount_var=1
fi

...

if [ "$umount_var" = "1" ]; then
    umount /var
fi
```
